gguf-py/gguf/metadata.py

```diff
@@ -0,0 +1,350 @@
+                    metadata.name = Metadata.id_to_title(model_full_name_component)
+                if metadata.organization is None and org_component is not None:
+                    metadata.organization = Metadata.id_to_title(org_component)
+                if metadata.basename is None and basename is not None:
+                    metadata.basename = basename
+                if metadata.finetune is None and finetune is not None:
+                    metadata.finetune = finetune
+                if metadata.version is None and version is not None:
+                    metadata.version = version
+                if metadata.parameter_class_attribute is None and parameter_class_attribute is not None:
+                    metadata.parameter_class_attribute = parameter_class_attribute
+                if metadata.source_url is None and org_component is not None and model_full_name_component is not None:
+                    metadata.source_url = f"https://huggingface.co/{org_component}/{model_full_name_component}"
+                if metadata.source_hf_repo is None and org_component is not None and model_full_name_component is not None:
+                    metadata.source_hf_repo = f"{org_component}/{model_full_name_component}"
+
+        # Directory Folder Name Fallback Heuristics
+        ############################################
+        if model_path is not None:
+            model_id = model_path.name
+            model_full_name_component, org_component, basename, finetune, version, parameter_class_attribute = Metadata.get_model_id_components(model_id)
+            if metadata.name is None and model_full_name_component is not None:
+                metadata.name = Metadata.id_to_title(model_full_name_component)
+            if metadata.organization is None and org_component is not None:
+                metadata.organization = Metadata.id_to_title(org_component)
+            if metadata.basename is None and basename is not None:
+                metadata.basename = basename
+            if metadata.finetune is None and finetune is not None:
+                metadata.finetune = finetune
+            if metadata.version is None and version is not None:
+                metadata.version = version
+            if metadata.parameter_class_attribute is None and parameter_class_attribute is not None:
+                metadata.parameter_class_attribute = parameter_class_attribute
+            if metadata.source_url is None and org_component is not None and model_full_name_component is not None:
+                metadata.source_url = f"https://huggingface.co/{org_component}/{model_full_name_component}"
+            if metadata.source_hf_repo is None and org_component is not None and model_full_name_component is not None:
+                metadata.source_hf_repo = f"{org_component}/{model_full_name_component}"
+
+        return metadata
+
+
+class TestStringMethods(unittest.TestCase):
```

Comment:
Tests should have their own module or at the very least execute within the same module.